src/utils/process.rs

```diff
@@ -854,9 +859,8 @@ pub mod tests {
     fn test_process_testing_assert_never_used() {
         let _args = FakeParentArgs::once("never used");
 
-        // causes a panic while panicking, so can't test:
-        // let _args = FakeParentArgs::for_scope(&"never used");
-        // let _args = FakeParentArgs::once(&"never used");
+        let _args = FakeParentArgs::for_scope(&"never used");
+        let _args = FakeParentArgs::once(&"never used");
     }
 
```

Comment:
Nice. Now that this is testable, this test needs to be split into separate cases. `once()` should panic on drop, even stand-alone as before, because it is never used.

But calling `for_scope` twice should panic in the constructor because the previous state (in the thread local variable) was not cleared out properly (and previously the drop also panicked because it encountered bad state as well).

Actually, couldn't `FakeParentArgs::error` just set the state so that drop doesn't panic again? And to ensure the correct panic is hit maybe the more precise [#[should_panic(expected = "msg here")]](https://doc.rust-lang.org/book/ch11-01-writing-tests.html#checking-for-panics-with-should_panic) can be used.

Reply:
I think all `FakeParentArgs` will go through `drop` before `error` being called in this construct.

```
            if FAKE_ARGS.with(|a| a.replace(initial(string_vec))) != TlsState::None {
                Self::error(from_);
            }
```
Thus, we'll only get panics from the `drop` method and none from the constructors